src/Utility/Hash.php

```diff
@@ -300,12 +300,10 @@ public static function insert(array $data, $path, $values = null)
 
         foreach ($data as $k => $v) {
             if (static::_matchToken($k, $token)) {
-                if ($conditions && static::_matches($v, $conditions)) {
-                    $data[$k] = array_merge($v, $values);
-                    continue;
-                }
-                if (!$conditions) {
-                    $data[$k] = static::insert($v, $nextPath, $values);
+                if (!$conditions || static::_matches($v, $conditions)) {
+                    $data[$k] = $nextPath
+                        ? static::insert($v, $nextPath, $values)
+                        : array_merge($v, (array) $values);
```

Comment:
This causes a [PHPCS error](https://travis-ci.org/cakephp/cakephp/jobs/71939961).